### PR TITLE
Server implementation

### PR DESCRIPTION
**Pull request by o-rumiantsev**

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -23,23 +36,334 @@ const transports = {
...
+        }
+      });
+    }
+
+    const definition = {};
+
+    interfaceNames.forEach(api => {
+      definition[api] = [];
+      const methods = Object.keys(interfaces[api]);
+      methods.forEach(
```
Comment (mille-nium) [inline lib/server.js]: Unnecessary `forEach` here, simply ```definition[api] = methods```

File: lib/server.js [hunk 2]
```diff
@@ -23,23 +36,340 @@ const transports = {
...
+  listInterfaces: interfaces => {
+    const schema = {};
+
+    for (const interfaceName in interfaces) {
+      schema[interfaceName] = {};
+      schema[interfaceName].methods = [];
+
+      const { description } = interfaces[interfaceName];
+
+      if (description) {
```
Comment (SemenchenkoVitaliy) [inline lib/server.js]: You can make it single-liner

File: lib/server.js [hunk 3]
```diff
@@ -23,23 +36,340 @@ const transports = {
...
+  //            ...
+  //          }
+  //          where `methodNames` is an array of names of
+  //          interface methods
+  //
+  selectApplication: (interfaces, interfaceNames) => {
+    const existingInterfaces = Object.keys(interfaces);
+    const allInterfaces = interfaceNames[0] === '*';
+
+    if (allInterfaces) {
```
Comment (SemenchenkoVitaliy) [inline lib/server.js]: Same here

File: lib/server.js [hunk 4]
```diff
@@ -23,23 +36,340 @@ const transports = {
...
+// - call - Metacom call object
+// - apps - Metacom.Server applications index
+//
+// Returns: tuple [error, result]
+//
+const inspection = (call, apps) => {
+  const { version, methodName, args } = call;
+  const fn = inspectionMethods[methodName];
+
+  if (methodName === 'listApplications') {
```
Comment (SemenchenkoVitaliy) [inline lib/server.js]: Same here, like:
```js
if (methodName === 'listApplications') return [null, fn(apps)];
```

File: lib/server.js [hunk 5]
```diff
@@ -23,23 +36,340 @@ const transports = {
...
+    return [null, result];
+  }
+
+  const app = apps.get(call.application);
+  if (!app) return [APP_NOT_FOUND];
+
+  const interfaces = app.get(version).api;
+
+  if (methodName === 'selectApplication') {
+    const interfaceNames = args;
```
Comment (SemenchenkoVitaliy) [inline lib/server.js]: I think you can do
```js
return fn(interfaces, args);
```

File: lib/server.js [hunk 6]
```diff
@@ -23,23 +36,340 @@ const transports = {
...
+    const [error, result] = fn(interfaces, interfaceNames);
+    return [error, result];
+  } else if (methodName === 'listInterfaces') {
+    const result = fn(interfaces);
```
Comment (SemenchenkoVitaliy) [inline lib/server.js]: ditto

**Comment by tshemsedinov** [comment]
Closed in favour of new version 0.1.x implementing new specification

**Pull request closed without merging**